### PR TITLE
Update spec conformance tests with object and binding pattern changes

### PR DESCRIPTION
**Pull request by Anoukh**
## Purpose
Update spec conformance tests with object and binding pattern changes

### CODE REVIEW

File: tests/ballerina-spec-conformance-tests/S05_behavioral_values/tests/objects/abstract_client_object_declaration.bal [hunk 1]
```diff
@@ -169,3 +169,38 @@ function testAbstractClientObjectDeclaration() {
...
+};
+
+type ObjReferenceToClientAbstractObject client object {
+    *ClientAbstractObject;
+
+    private float privateFloatField;
+
+    function __init() {
+        self.publicStringField = "string";
+        self.floatField = 3.0;
```
Comment (MaryamZi) [inline tests/ballerina-spec-conformance-tests/S05_behavioral_values/tests/objects/abstract_client_object_declaration.bal]: Shall we use constants? O:)

File: tests/ballerina-spec-conformance-tests/S05_behavioral_values/tests/objects/abstract_client_object_declaration.bal [hunk 2]
```diff
@@ -169,3 +169,38 @@ function testAbstractClientObjectDeclaration() {
...
+    remote function getPrivateField() returns float {
+        return self.privateFloatField;
+    }
+};
+
+@test:Config {}
+function testClientAbstractObjectDeclaration() {
+    ObjReferenceToClientAbstractObject abstractClientObj = new();
+
+    test:assertTrue(abstractClientObj.publicStringField == "string",
```
Comment (MaryamZi) [inline tests/ballerina-spec-conformance-tests/S05_behavioral_values/tests/objects/abstract_client_object_declaration.bal]: Can't we use `assertEquals` instead? Same for other `==` checks.

File: tests/ballerina-spec-conformance-tests/S05_behavioral_values/tests/objects/client_object_declaration.bal [hunk 3]
```diff
@@ -126,7 +126,7 @@ private remote function ClientObject.privateRemoteMethodDecl(string argOne, int
 }
 
 remote function ClientObject.defaultVisibiltyRemoteMethodDecl(string argOne, int argTwo) {
```
Comment (MaryamZi) [inline tests/ballerina-spec-conformance-tests/S05_behavioral_values/tests/objects/client_object_declaration.bal]: Should we rename these functions from `XxxDecl` to `XxxDefn` ?

File: tests/ballerina-spec-conformance-tests/S05_binding_patterns_and_variables/tests/record_patterns/record_typed_binding_pattern.bal [hunk 4]
```diff
@@ -106,6 +106,29 @@ function testRecordTypedBindingPatternWithClosedRestParam() {
     test:assertEquals(field3, 100.1, msg = EXPECTED_RECORD_DESTRUCTURE_TO_VAR_DEF_FAILURE_MESSAGE);
 }
 
+@test:Config {}
+function testRecordTypedBindingPatternWithOnlyRestParam() {
+    BindingPattern bindingPattern = {
+        field1: 11,
```
Comment (MaryamZi) [inline tests/ballerina-spec-conformance-tests/S05_binding_patterns_and_variables/tests/record_patterns/record_typed_binding_pattern.bal]: Again, shall we use constants or even separate variables?

File: tests/ballerina-spec-conformance-tests/S05_behavioral_values/tests/objects/client_object_declaration.bal (hunk 3 above)
Comment (Anoukh) [reply tests/ballerina-spec-conformance-tests/S05_behavioral_values/tests/objects/client_object_declaration.bal]: This is an outside implementation of a method declaration. Hence the name XxxDecl

Comment (MaryamZi) [reply tests/ballerina-spec-conformance-tests/S05_behavioral_values/tests/objects/client_object_declaration.bal]: Got it.

**Pull request merged**